File: stress/t/flt_horner.f
      PROGRAM FHORNER
      IMPLICIT DOUBLE PRECISION (A-H,O-Z)
      INTEGER*8 I,N
      DIMENSION COEF(0:7)
      COEF(0)=1.0D0
      COEF(1)=-(3.0D0)
      COEF(2)=2.0D0
      COEF(3)=0.5D0
      COEF(4)=-(1.25D0)
      N=5
      X=1.7D0
      ACC=0.0D0
      I=N-1
   10 IF (I.LT.0) GOTO 20
        C=COEF(I)
        ACC=ACC*X+C
        I=I-1
        GOTO 10
   20 CONTINUE
      WRITE(*,'(A,F0.6)') 'p= ',ACC
      END

File: stress/t/flt_horner.c
#include <stdio.h>
#include <math.h>
int main(void){
  long i,n=5; double x=1.7,acc,c; double coef[8];
  coef[0]=1.0; coef[1]=-(3.0); coef[2]=2.0; coef[3]=0.5; coef[4]=-(1.25);
  acc=0.0;
  for(i=n-1;i>=0;i--){ c=coef[i]; acc=acc*x+c; }
  printf("p= %f\n",acc);
  return 0;
}
